packages/mgt-chat/src/statefulClient/StatefulGraphChatListClient.ts

```diff
@@ -207,22 +224,79 @@ class StatefulGraphChatListClient implements StatefulClient<GraphChatListClient>
+        // update the user list to remove the user; while we could add a "User removed" message
+        // the Teams client doesn't show a message and the Graph API does not show a message either
+        const membersToRemove = (event.message.eventDetail as MembersDeletedEventMessageDetail).members;
+        if (membersToRemove) {
+          const membersToRemoveSet = new Set(membersToRemove.map(member => member.id));
+          chatThread.members = chatThread.members?.filter(member => {
+            const user = member as AadUserConversationMember;
+            return !membersToRemoveSet.has(user.userId);
+          });
         }
+      } else if (event.type === 'teamsAppInstalled' && event.message?.eventDetail && chatThread) {
+        // add a message about the app being added; Teams does "User added", but this provides
+        // more clarity since the user list doesn't change
+        const details = event.message.eventDetail as TeamsAppInstalledEventMessageDetail;
+        chatThread.lastMessagePreview = {
+          body: {
+            content: `${details.teamsAppDisplayName || details.teamsAppId} app added`,
+            contentType: 'text'
+          } as ItemBody,
+          lastModifiedDateTime: event.message.lastModifiedDateTime
+        } as ChatMessageInfo;
+        log('injected', chatThread.lastMessagePreview);
+        bringToTop();
+      } else if (event.type === 'teamsAppRemoved' && event.message?.eventDetail && chatThread) {
+        // there is no behavior in teams for an app being removed
+      } else if (event.type === 'chatMessageEdited' && chatThread) {
+        // update the last message preview in-place
+        chatThread.lastMessagePreview = event.message as ChatMessageInfo;
+        chatThread.lastUpdatedDateTime = event.message.lastModifiedDateTime;
+      } else if (event.type === 'chatMessageDeleted' && chatThread) {
+        // update the last message preview in-place
+        chatThread.lastMessagePreview = {
+          lastModifiedDateTime: event.message.lastModifiedDateTime,
+          isDeleted: true
+        } as ChatMessageInfo;
+      } else if (event.type === 'chatMessageReceived' && event.message && chatThread) {
+        // update the last message preview and bring to the top
+        chatThread.lastMessagePreview = event.message as ChatMessageInfo;
+        chatThread.lastUpdatedDateTime = event.message.lastModifiedDateTime;
+        bringToTop();
+      } else if (event.type === 'chatMessageReceived' && event.message?.chatId) {
+        // create a new chat thread at the top
+        const newChatThread: GraphChat = {
+          id: event.message.chatId,
+          lastMessagePreview: event.message as ChatMessageInfo,
+          lastUpdatedDateTime: event.message.lastModifiedDateTime
+        };
+        draft.chatThreads.unshift(newChatThread);
+      } else {
+        error(`received unrecognized event type '${event.type}' from the user subscription.`);
```

Comment:
i mean... is it really an error? it not an error per se right?

Reply:
fair, I can change that to log.